lib/features/push_notification/data/repository/fcm_repository_impl.dart

```diff
@@ -104,4 +110,26 @@ class FCMRepositoryImpl extends FCMRepository {
   Future<bool> destroySubscription(String subscriptionId) {
     return _fcmDatasource[DataSourceType.network]!.destroySubscription(subscriptionId);
   }
+
+  @override
+  Future<List<PresentationMailbox>> getMailboxesNotPutNotifications(Session session, AccountId accountId) async {
+    final mailboxesCache = await _mapMailboxDataSource[DataSourceType.local]!.getAllMailboxCache();
+    final mailboxesCacheNotPutNotifications = mailboxesCache
+      .map((mailbox) => mailbox.toPresentationMailbox())
+      .where((presentationMailbox) => presentationMailbox.pushNotificationDeactivated)
+      .toList();
+    log('FCMRepositoryImpl::getMailboxesNotPutNotifications():mailboxesCacheNotPutNotifications: $mailboxesCacheNotPutNotifications');
+    if (mailboxesCacheNotPutNotifications.isNotEmpty && mailboxesCacheNotPutNotifications.length == 5) {
+      return mailboxesCacheNotPutNotifications;
```

Comment:
Why did you use the condition `mailboxesCacheNotPutNotifications.length == 5`? Let me knowwww ạaa

Reply:
can you write the ADR?

Reply:
> can you write the ADR?

Updated!. Please see below.